app/master/cluster_master.py

```diff
@@ -279,7 +282,7 @@ def handle_result_reported_from_slave(self, slave_url, build_id, subjob_id, payl
         build = self._all_builds_by_id[int(build_id)]
         slave = self._all_slaves_by_url[slave_url]
         # If the build has been canceled, don't work on the next subjob.
-        if not build.is_finished:  # WIP(joey): This check should be internal to the Build object.
+        if not build.is_finished or build.has_error:  # WIP(joey): This check should be internal to the Build object.
```

Comment:
yeah. Visited the /v1/slave endpoint and current_build_id was null for both.

I made the following setup_build file
```python
import time
from os import system
import sys

if system('mkdir lockdir') == 0:
    sys.exit(1)
else:
    time.sleep(5)
    sys.exit(0)
```

The first time it's run the `mkdir` exits 0. The second time the `mkdir` fails.